scripts/factory-common-install/libexec/app_current_device_vnc.sh

```diff
@@ -0,0 +1,167 @@
+  read_or_prompt_for_current_device__hostname "vnc_hostname"
+  read_or_prompt_for_current_device__ssh_port "device_ssh_port"
+
+  cleanup() {
+    local vnc_user="${1?}"
+    local vnc_hostname="${2?}"
+    local device_ssh_port="${3?}"
+    local vnc_remote_port="${4?}"
+    local systemd_service_name="${5?}"
+    shift 5
+
+    echo "Cleanup required"
+    if is_x11vnc_systemd_process_stopped "${vnc_user}" "${vnc_hostname}" "${device_ssh_port}" "${systemd_service_name}"; then
+        echo "x11vnc systemd process is already closed"
+    else
+        echo "Closing x11vnc systemd process"
+        close_x11vnc_systemd_process "${vnc_user}" "${vnc_hostname}" "${device_ssh_port}" "${systemd_service_name}"
+    fi
+
+    if is_x11vnc_pid_stopped "${vnc_user}" "${vnc_hostname}" "${device_ssh_port}" "${vnc_remote_port}"; then
+        echo "x11vnc pid is already stopped"
+    else
+        echo "Closing x11vnc pid"
+        exit_x11vnc "${vnc_user}" "${vnc_hostname}" "${device_ssh_port}"
+    fi
+
+    return 0
+  }
+  trap "cleanup '${vnc_user}' '${vnc_hostname}' '${device_ssh_port}' '${vnc_remote_port}' '${systemd_service_name}'" EXIT SIGINT SIGQUIT
+
+  user_info="$(get_factory_info__user_full_name)<$(get_factory_info__user_email)>"
+  vnc_passwd="$(pwgen -cn 32 | head -n1)"
+
+  # Validation that the service that we are about to launch does not exist yet.
+  # if the service is alive we ask the user if it wants to kill the existing session.
+  echo " -> Making sure x11vnc is not aready started"
+  echo '"${vnc_user}" "${vnc_hostname}" "${device_ssh_port}" "${systemd_service_name}"'
+  if ! is_x11vnc_systemd_process_stopped "${vnc_user}" "${vnc_hostname}" "${device_ssh_port}" "${systemd_service_name}"; then
+    active_session_info=$(ssh "${vnc_user}@${vnc_hostname}" -p "${device_ssh_port}" "systemctl -q --no-pager --user status '${systemd_service_name}'") && true
+    echo "An active x11vnc session has been detected"
+    echo "Here is some information about the active session"
+    echo "${active_session_info}"
+    echo ""
+    read -n 1 -p 'Do you want to close the session? (Y)es or (N)o' user_input
+    echo ""
+    case $user_input in
+      [yY] )
+        echo "Okay, we will kill the existing session"
+        cleanup "${vnc_user}" "${vnc_hostname}" "${device_ssh_port}" "${vnc_remote_port}" "${systemd_service_name}"
+        # Sleep 5 seconds to avoid getting kicked out by the other script cleanup.
+        sleep 5
+        ;;
+      [nN] )
+        echo "Okay, leaving the existing session as is."
+        trap - EXIT SIGINT SIGQUIT
+        return 0
+        ;;
+      * )
+        echo "Erroneous answer. Doing nothing."
+        trap - EXIT SIGINT SIGQUIT
+        return 0
+        ;;
+    esac
+  fi
+
+  # x11vnc is launched as a connect once server, as soon as vncviewer closes x11vnc also closes.
+  echo " -> Starting x11vnc"
+  cmd="$(cat <<EOF
+systemd-run \
+--remain-after-exit \
+--property=Type=forking \
```

Comment:
Ok. I understand that you use the forking type because of the x11vnc -bg arg which daemonize the service. Right?

[systemd.service](https://www.freedesktop.org/software/systemd/man/systemd.service.html#Examples)

Reply:
That is correct.